### DISCUSSION TITLE
[Proposal] introduce a sub project about Graph-AI for HugeGraph

### DISCUSSION THREAD

**Discussion by simon824** [apache/incubator-hugegraph#2318] [Ideas]
Dear all,

Graph intelligence has seen significant growth and popularity in recent years, with many applications emerging in different industries. This field is closely connected to HugeGraph, which serves as a strong foundation for storing, analyzing, and processing graph-based data.

I suggest creating a new Python repository for artificial intelligence related to HugeGraph. This repository will include applications and integrations of HugeGraph with large language models, graph machine learning, graph neural networks, graph embeddings, and more.

By establishing this repository, our goal is to explore the integration of HugeGraph with various AI frameworks, providing a comprehensive resource for AI practitioners to utilize the power of HugeGraph in their projects.

The repository will consist of the following sub-modules:

1. hugegraph-llm:
This module will house the implementation and research related to large language models. It will include runnable demos and can also be used as a third-party library, reducing the cost of using graph systems and the complexity of building knowledge graphs. Graph systems can help large models address challenges like timeliness and hallucination, while large models can assist graph systems with cost-related issues. Therefore, this module will explore more applications and integration solutions for graph systems and large language models. We have already integrated with the LLM application frameworks [LangChain](https://github.com/apache/incubator-hugegraph/discussions/2251) and Llama-Index, and will provide usage examples in this module.

2. hugegraph-ml:
This module will focus on integrating HugeGraph with graph machine learning, graph neural networks, and graph embeddings libraries. It will build an efficient and versatile intermediate layer to seamlessly connect with third-party graph-related ML frameworks.

3. hugegraph-python-client/computer-client:
The hugegraph-python-client is a Python client for HugeGraph. It is used to define graph structures and perform CRUD operations on graph data. Both the hugegraph-llm and hugegraph-ml modules will depend on this foundational library. Additionally, the client will be integrated with HugeGraph-Computer. Graph computation is often necessary for graph learning and preprocessing graph data. Developers can significantly improve processing efficiency by leveraging HugeGraph-Computer.

If anyone has any ideas or questions about this proposal, please leave a comment here for discussion.

- https://github.com/apache/incubator-hugegraph-ai

**Comment by diaohancai** [comment]
### Sounds great!

**Comment by javeme** [comment]
Looks good.

For the convenience of adapt-adapt and release, we may consider to placed the Python client in the toolchain repo like: https://github.com/apache/incubator-hugegraph-toolchain/tree/master/hugegraph-client

**Reply by imbajin** [reply]
Both fine to me, **python-based** repo can be easily debugged and modified together (in the beta stage)
Maybe we could consider the early state in the AI repository for `BETA`, then migrate to the toolchain for `release`?

**Reply by javeme** [reply]
I think it's ok
